src/autobot.ts

```diff
@@ -111,16 +113,41 @@ export class Autobot {
 
   private initializePlugins(scope: ExecutionScope, context?: PRContext) {
     const scopePluginInstances = <T extends Plugin>() =>
-      this.plugins.filter(plugin => plugin.scope === scope).map(Plugin => new Plugin()) as T[];
+      Object.values(this.plugins)
+        .filter((options): options is UninitializedPlugin => !options.initialized)
```

Comment:
This is a ts thing I learned tonight to get `filter` to return the type you want it to. 

That's the `:options is UninitializedPlugin` part. 

I really wish TS could handle this better generally. It was discussed [here](https://github.com/Microsoft/TypeScript/issues/20812)

cc @orta 😉